raft.go

```diff
@@ -601,9 +601,7 @@ func (r *raft) send(m pb.Message) {
 // Returns true if a message was sent, or false otherwise. A message is not sent
 // if the follower log and commit index are up-to-date, the flow is paused (for
 // reasons like in-flight limits), or the message could not be constructed.
-func (r *raft) maybeSendAppend(to uint64) bool {
-	pr := r.trk.Progress[to]
-
+func (r *raft) maybeSendAppend(to uint64, pr *tracker.Progress) bool {
```

Comment:
What's the intention to pass the `tracker.Progress`? Save a couple of cpu cycles? If yes, then I think readability outweighs the performance here. The lookup in map should be very fast.

Reply:
Yes, just "normalizing" the code here. All the callers of `maybeSendAppend` already have a `Progress`. Will remove this commit though.